open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventProducer.java

```diff
@@ -97,58 +96,97 @@ private void publishEvent(String event) throws ConnectorCheckedException
+                ProducerRecord<String, String> record = new ProducerRecord<>(topicName, localServerId, event);
+                producer.send(record).get();
+                eventSent = true;
+                messageSendCount++;
+            }
+            catch (ExecutionException error)
+            {
+                /*
+                 * This may be a simple timeout or something else more
+                 */
+                log.debug("Kafka had trouble sending event: " + event + "exception message is " + error.getMessage());
+
+                if( !isExceptionRetryable(error)) {
+                    /* kafka thinks this isn't a retryable problem */
+                    /* so let the caller try */
+                    log.error("Exception in sendEvent " + error.toString());
+                    KafkaOpenMetadataTopicConnectorErrorCode errorCode = KafkaOpenMetadataTopicConnectorErrorCode.ERROR_SENDING_EVENT;
+                    String errorMessage = errorCode.getErrorMessageId() + errorCode.getFormattedErrorMessage(error.getClass().getName(),
+                            topicName,
+                            error.getMessage());
+
+                    throw new ConnectorCheckedException(errorCode.getHTTPErrorCode(),
+                            this.getClass().getName(),
+                            methodName,
+                            errorMessage,
+                            errorCode.getSystemAction(),
+                            errorCode.getUserAction(),
+                            error);
+                }
+                if (eventRetryCount == 10)
+                {
+                    /* we've retried now let the caller retry */
+                    break;
+                }
+                else
+                {
+                    if (eventRetryCount == 0)
+                    {
+                        KafkaOpenMetadataTopicConnectorAuditCode auditCode;
+
+                        auditCode = KafkaOpenMetadataTopicConnectorAuditCode.EVENT_SEND_IN_ERROR_LOOP;
+                        auditLog.logRecord(methodName,
+                                           auditCode.getLogMessageId(),
+                                           auditCode.getSeverity(),
+                                           auditCode.getFormattedLogMessage(topicName,
+                                                                            Long.toString(messageSendCount),
+                                                                            Long.toString(this.getSendBufferSize()),
+                                                                            error.getMessage()),
+                                           null,
+                                           auditCode.getSystemAction(),
+                                           auditCode.getUserAction());
+                    }
+
+                    eventRetryCount++;
+                }
+            }
+            catch (WakeupException error)
```

Comment:
Though this occurs at shutdown, could it occur in other situations too? 
For example another thread could use wakeup() against this thread - though I think we do?
I'm wondering if we should pause and retry ... though at shutdown I'd be concerned we wait too long
If we really only think wakeup will be at shutdown then this handling is ok-ish, but it's one route out of the method where we don't log an audit message at all, which probably would be desirable?

Reply:
@planetf1 This code is actually original code that was removed as part of this first attempt to fix this issue. I put the behaviour back after getting feedback on the first fix. There is nothing the producer is doing that would cause kafka to throw this.

Reply:
Good to stick with this for our immediate kafka fix - can consider revisiting with a new connector implementation